realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -708,17 +708,35 @@ bool BaseRealSenseNode::setBaseTime(double frame_time, rs2_timestamp_domain time
     return false;
 }
 
+uint64_t BaseRealSenseNode::millisecondsToNanoseconds(double timestamp_ms)
+{
+        // modf breaks input into an integral and fractional part
+        double int_part_ms, fract_part_ms;
+        fract_part_ms = modf(timestamp_ms, &int_part_ms);
+
+        //convert both parts to ns
+        static constexpr uint64_t milli_to_nano = 1000000;
+        uint64_t int_part_ns = static_cast<uint64_t>(int_part_ms) * milli_to_nano;
+        // for fract_part_ns, multiplie ns * 10, then divide by 10 while rounding
+        // to be aligned with librealsense get_timestamp API
+        uint64_t fract_part_ns = static_cast<uint64_t>(fract_part_ms * milli_to_nano * 10);
+        fract_part_ns = (fract_part_ns + 5)/10;
```

Comment:
Looking at this now it seems like a redundant division.
Use `static_cast<uint64_t>( ( fract_part_ms + 0.5 ) * milli_to_nano )` to eliminate it.

Reply:
Or just `std::round` that does the same. It is more readable.

Reply:
After consulting with StackOverflow I suggest you use `round` as the `+0.5` method does not handle all corner cases (Inf, NaN and such).

Reply:
done